src/main/java/io/github/thebusybiscuit/slimefun4/implementation/accessibility/PlayerAccessManagerImpl.java

```diff
@@ -0,0 +1,247 @@
+        reset();
+        if (!SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_14)) {
+            // This feature was introduced post 1.13. Cannot load newer data on an older version
+            return;
+        }
+        final ItemMeta meta = itemStack.getItemMeta();
+        if (meta != null) {
+            load(namespace, meta);
+        }
+
+    }
+
+    @Override
+    public void load(@Nonnull final NamespacedKey namespace, @Nonnull final Block block) {
+        reset();
+        if (!SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_14)) {
+            return;
+        }
+        final BlockState state = block.getState();
+        if (state instanceof TileState) {
+            load(namespace, ((TileState) state));
+        }
+    }
+
+    @Override
+    public void saveTo(@Nonnull NamespacedKey namespace, @Nonnull final ItemStack item) {
+        if (!SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_14)) {
+            // This feature was introduced post 1.13. Cannot save.
+            return;
+        }
+        final ItemMeta meta = item.getItemMeta();
+        if (meta != null) {
+            saveTo(namespace, meta);
+            item.setItemMeta(meta);
+        }
+    }
+
+    @Override
+    public void saveTo(@Nonnull final NamespacedKey namespace, @Nonnull final Block block) {
+        if (!SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_14)) {
+            return;
+        }
+        final BlockState state = block.getState();
+        if (state instanceof TileState) {
+            saveTo(namespace, ((TileState) state));
+            state.update(true, false);
+        }
+    }
+
+    @Override
+    public void loadFromJsonElement(@Nonnull final JsonElement element) throws IllegalArgumentException {
+        reset();
+        if (!element.isJsonObject()) {
+            return;
+        }
+        for (Map.Entry<String, JsonElement> entry : element.getAsJsonObject().entrySet()) {
+            final String rawClass = entry.getKey();
+            final Class<?> clazz;
+            try {
+                clazz = Class.forName(rawClass);
+            } catch (ClassNotFoundException ex) {
+                Slimefun.getLogger()
+                    .log(Level.WARNING, "Invalid AccessData! Unknown AccessData class: " + rawClass);
+                continue;
+            }
+            if (!AccessData.class.isAssignableFrom(clazz)) {
+                Slimefun.getLogger()
+                    .log(Level.WARNING, "Invalid AccessData! AccessData is not assignable from: " + rawClass);
+                continue;
+            }
+            final Optional<?> optional =
+                SlimefunPlugin.getJsonDeserializationService().deserialize(clazz, entry.getValue());
+            optional.ifPresent((o) -> {
+                AccessData<?> accessData = (AccessData<?>) o;
+                dataMap.put(accessData.getType(), accessData);
```

Comment:
Formatting

Reply:
could you clarify why this is poorly formatted, I've read the updated code-style guide but I may have missed / misunderstood certain points I guess